config/management/genesis/src/validator_config.rs

```diff
@@ -43,13 +51,33 @@ impl ValidatorConfig {
             self.owner_name,
         )?;
 
+        // Fetch keys from storage
+        let (consensus_key, fullnode_network_key, validator_network_key) =
+            get_keys_from_storage(storage_name, &self.validator_backend.validator_backend)?;
+
+        // Append ln-noise-ik and ln-handshake protocols to base network addresses
+        // and encrypt the validator address.
+        let raw_enc_validator_address = encode_validator_address(
+            validator_network_key,
+            self.validator_address,
+            &owner_account,
+            sequence_number,
+            &TEST_SHARED_VAL_NETADDR_KEY,
+            TEST_SHARED_VAL_NETADDR_KEY_VERSION,
+            idx_addr,
+        )?;
+
+        let raw_fullnode_address =
+            encode_fullnode_address(fullnode_network_key, self.fullnode_address)?;
+
         // Create the validator config script for the validator node
         let validator_config_script = create_validator_config_script(
-            &self.validator_backend,
+            consensus_key,
+            fullnode_network_key,
+            validator_network_key,
             owner_account,
-            self.validator_address,
-            self.fullnode_address,
-            sequence_number,
+            raw_enc_validator_address,
+            raw_fullnode_address,
```

Comment:
this feels like something that should happen in create_validator_config_script, if we do the structopt flatten approach I mentioned earlier.